pegasus/helpers/section_api_helpers.rb

```diff
@@ -252,6 +252,14 @@ def self.valid_courses(user_id = nil)
         end
   end
 
+  def self.progress_disabled_courses(user_id = nil)
+    disabled_courses = valid_courses(user_id).select do |course|
+      script_name = Script.find(course[:id]).name
+      !Gatekeeper.allows('postMilestone', where: {script_name: script_name}, default: true)
+    end
```

Comment:
We need to be able to know whether to show this message or not on a per course basis. This means we need to get the gatekeeper settings for each course on page load. Not sure if this ends up being more querying than we'd like.